NEXTCLOUD ANDROID CRASH REPORT

### Steps to reproduce
1. 
2. 
3. 

### Expected behaviour
- Tell us what should happen
Login into to Nextcloud Android App with username password of a registered user. 
Grant Access to your Android device to access you XXXX nextcloud account. 
### Actual behaviour
- Tell us what happens
App Crashed
### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version:
Brand: OnePlus
Device: OnePlus8Pro
Model: IN2020
Id: RP1A.201005.001
Product: OnePlus8Pro
Device model: 

Stock or customized system:
Stock
Nextcloud app version:
3.18.0
Nextcloud server version:
22.2.0
Reverse proxy:

### Logs
#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
************ CAUSE OF ERROR ************

java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
Caused by: java.lang.RuntimeException: Accessing result data after operation failed!
	at com.owncloud.android.lib.common.operations.RemoteOperationResult.getResultData(RemoteOperationResult.java:503)
	at com.owncloud.android.authentication.AuthenticatorAsyncTask.doInBackground(AuthenticatorAsyncTask.java:80)
	at com.owncloud.android.authentication.AuthenticatorAsyncTask.doInBackground(AuthenticatorAsyncTask.java:45)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30180090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: OnePlus
Device: OnePlus8Pro
Model: IN2020
Id: RP1A.201005.001
Product: OnePlus8Pro

************ FIRMWARE ************
SDK: 30
Release: 11
Incremental: 2110091917

**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

--- a/src/main/java/com/owncloud/android/authentication/AuthenticatorAsyncTask.java
+++ b/src/main/java/com/owncloud/android/authentication/AuthenticatorAsyncTask.java
@@ -76,20 +76,24 @@ public class AuthenticatorAsyncTask extends AsyncTask<Object, Void, RemoteOperat
             RemoteOperationResult<UserInfo> userInfoResult = new GetUserInfoRemoteOperation().execute(nextcloudClient);
 
             // Operation - try credentials
-            OwnCloudClient client = OwnCloudClientFactory.createOwnCloudClient(uri, context, true);
-            client.setUserId(userInfoResult.getResultData().getId());
-            client.setCredentials(credentials);
+            if (userInfoResult.isSuccess()) {
+                OwnCloudClient client = OwnCloudClientFactory.createOwnCloudClient(uri, context, true);
+                client.setUserId(userInfoResult.getResultData().getId());
+                client.setCredentials(credentials);
 
-            ExistenceCheckRemoteOperation operation = new ExistenceCheckRemoteOperation(ROOT_PATH, SUCCESS_IF_ABSENT);
-            result = operation.execute(client);
+                ExistenceCheckRemoteOperation operation = new ExistenceCheckRemoteOperation(ROOT_PATH, SUCCESS_IF_ABSENT);
+                result = operation.execute(client);
 
-            if (operation.wasRedirected()) {
-                RedirectionPath redirectionPath = operation.getRedirectionPath();
-                String permanentLocation = redirectionPath.getLastPermanentLocation();
-                result.setLastPermanentLocation(permanentLocation);
+                if (operation.wasRedirected()) {
+                    RedirectionPath redirectionPath = operation.getRedirectionPath();
+                    String permanentLocation = redirectionPath.getLastPermanentLocation();
+                    result.setLastPermanentLocation(permanentLocation);
+                }
+
+                result.setResultData(userInfoResult.getResultData());
+            } else {
+                result = userInfoResult;
             }
-            
-            result.setResultData(userInfoResult.getResultData());
         } else {
             result = new RemoteOperationResult(RemoteOperationResult.ResultCode.UNKNOWN_ERROR);
         }

